### PR TITLE
fix: correct MDXPlugin type 

### PR DESCRIPTION
**Pull request by armano2**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Correct invalid types exposed for mdx related plugins

additionally this PR aligns versions of `unified` used in project

## Related issues/PRs

https://github.com/facebook/docusaurus/pull/6567
https://github.com/facebook/docusaurus/pull/8288

### CODE REVIEW

File: packages/docusaurus-mdx-loader/src/loader.ts [hunk 1]
```diff
@@ -61,15 +62,14 @@ const {
...
-  // @ts-expect-error: TODO
   remarkPlugins: [emoji, headings, toc],
   beforeDefaultRemarkPlugins: [],
   beforeDefaultRehypePlugins: [],
 };
 
 const compilerCache = new Map<string | Options, [Processor, Options]>();
 
-export type MDXPlugin = PluggableList;
+export type MDXPlugin = Pluggable;
```
Comment (armano2) [inline packages/docusaurus-mdx-loader/src/loader.ts:72]: this type was not correct as it was pointing to list of plugin instead of plugin itself

**Review by Josh-Cena — APPROVED** [review]
I'm good with this change, although the number of `@ts-expect-error`s is worrying ^^ This was the reason why I didn't bump `unified`.

**Comment by armano2** [comment]
i understand a reason, but issues with types started to popup as transitive dependencies where not up to date. `vFile` i new version has a different model

**Comment by Ralphhotrod** [comment]
OK  u do it

**Comment by slorber** [comment]
Will have to take a look at what happens but my mdx2 branch does not build anymore 😅 

https://github.com/facebook/docusaurus/pull/8288

**Comment by armano2** [comment]
@slorber build fails due to incorrect import after merge,

I used a little outdated version of your branch :)

```ts
import unified from 'unified';
```

this should be

```ts
import {unified} from 'unified';
```

**Comment by slorber** [comment]
Well, actually not so simple to fix due to ESM + Jest 🤪 

https://github.com/facebook/docusaurus/pull/8288/commits/2f6ff73b927ac08fcc9a45d1d3ae26aeec2f01d9

**Pull request merged**